msteams-platform/copilot/how-to-extend-copilot.md

```diff
@@ -0,0 +1,112 @@
+
+You can extend Microsoft 365 Copilot by building a plugin or by connecting to an external data source.
+
+## What is a plugin?
+
+A plugin allows Copilot to interact directly with third-party data, apps, and services, enhancing its capabilities and broadening its range of capabilities.
+Plugins allow Copilot to:
+
+* Retrieve real-time information, for example, latest news coverage on a product launch.
+* Retrieve knowledge-based information, for example, my team’s design files in Figma.
+* Perform actions on behalf of the user, for example, create a Jira ticket.
+
+## How do I build a plugin?
+
+> [!NOTE]
+> Plugins for Microsoft 365 Copilot are in early access preview. This documentation is subject to change. When you have access to [the developer preview program](https://aka.ms/plugins-dev-waitlist), you'll receive instructions on evaluating your message extension to be used by Microsoft 365 Copilot.
+
+If you already have a Teams message extension, then you already have a plugin that works with Copilot. If not, create a [message extension using the Teams Toolkit](~/messaging-extensions/what-are-messaging-extensions.md).
+
+**Scenario 1: I want to create a new plugin from an existing API or OpenAI plugin**
+
+[Join the early access program](https://aka.ms/PluginsEarlyAccess) to build a message extension that allows you to convert an existing API or OpenAI plugin to work for Microsoft Copilot.
+
+1. Using the Teams Toolkit for Microsoft Visual Studio Code, go to **Create plugin for Copilot**.
+2. Provide a link to either an OpenAPI specification or an OpenAI plugin.
+3. Update your manifest with plugin metadata such as name, publisher, APIs used.
+4. Optional: If you want your plugin to return an Adaptive Card, provide an Adaptive Card template and reference the template in the manifest.
+5. Test your plugin.
+6. Preview and publish your plugin.
+
+**Scenario 2: I want to build a plugin from an existing Microsoft 365 message extension**
+
+Your Message Extension will work with Copilot (after quality testing).
+
+## What makes a good plugin for Microsoft 365 Copilot?
+
+To ensure your plugin works as intended, it is important to include good descriptions in your manifest. This allows the underlying LLM to take full advantage of your plugin’s capabilities. It’s recommended to test different descriptions and instructions to find the most effective way for the LLM to utilize your plugin. Also, it’s important that the manifest file is up-to-date with any changes or enhancements you make to your plugin.
+
+1. App description: Include detailed descriptions of the app, supported scenarios, feature capabilities, and related keywords using the `shortDescription` and `longDescription` fields.
+
+   ```json
+   ... 
+   "name": { 
+   "shortDescription": "Work-item tracking and productivity app", 
```

Comment:
I noticed that these fields are not indented. Is it possible to do that?

Reply:
Updated.